internal/nfs/controller/volume.go

```diff
@@ -283,18 +290,18 @@ func (nv *NFSVolume) getNFSCluster() (string, error) {
 // setNFSCluster stores the NFS-cluster name in the CephFS journal.
 func (nv *NFSVolume) setNFSCluster(clusterName string) error {
 	if !nv.connected {
-		return fmt.Errorf("can not set NFS-cluster for %q: not connected", nv)
+		return fmt.Errorf("can not set NFS-cluster for %q: %w", nv, ErrNotConnected)
 	}
 
 	fs := fscore.NewFileSystem(nv.conn)
 	fsName, err := fs.GetFsName(nv.ctx, nv.fscID)
 	if err != nil {
-		return fmt.Errorf("failed to get filesystem name for ID %x: %w", nv.fscID, err)
+		return fmt.Errorf("filesystem name for ID %x %w: %v", nv.fscID, ErrNotFound, err)
 	}
 
 	mdPool, err := fs.GetMetadataPool(nv.ctx, fsName)
 	if err != nil {
-		return fmt.Errorf("failed to get metadata pool for %q: %w", fsName, err)
+		return fmt.Errorf("metadata pool for %q %w: %v", fsName, ErrNotFound, err)
```

Comment:
same here for error handling we should return `ErrNotFound` for pool not found error not for all the error cases isn't it?

Reply:
This is still not addressed yet?

Reply:
hmm, I thought it was? it seems I missed this one

Reply:
It should be addressed now too.